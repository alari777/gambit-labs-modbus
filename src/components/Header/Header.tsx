import { FC } from 'react';
import styles from '@/components/Header/Header.module.scss';
import Link from 'next/link';
import { useUser } from '@auth0/nextjs-auth0/client';

const Header: FC = () => {
  const { user, error, isLoading } = useUser();

  if (isLoading) return <div>Loading...</div>;
  if (error) return <div>{error.message}</div>;
  return (
    <header className={styles.header}>
      <nav className='navbar navbar-expand-lg navbar-light'>
        <div className='collapse navbar-collapse' id='navbarNavAltMarkup'>
          <div className='navbar-nav'>
            <Link
              data-testid='nav-home'
              className='nav-item nav-link active'
              href='/'
            >
              Home
            </Link>
            {user && (
              <>
                <Link
                  data-testid='nav-protection-page'
                  className='nav-item nav-link'
                  href='/protection-page'
                >
                  Protection Home Page
                </Link>
                <Link
                  data-testid='nav-logout'
                  className='nav-item nav-link'
                  href='/api/auth/logout'
                >
                  Logout ({user.name})
                </Link>
              </>
            )}
            {!user && (
              <Link
                data-testid='nav-login'
                className='nav-item nav-link'
                href='/api/auth/login'
              >
                Login
              </Link>
            )}
          </div>
        </div>
      </nav>
    </header>
  );
};

export default Header;
