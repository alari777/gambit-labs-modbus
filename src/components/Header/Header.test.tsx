import { render, screen } from '@testing-library/react';
import Header from './Header';

describe('should render header', () => {
  beforeEach(() => {
    render(<Header />);
  });

  test('should render home link in header navbar', () => {
    const navHome = screen.getByTestId('nav-home');
    expect(navHome).toBeInTheDocument();
    expect(navHome).toMatchSnapshot();
  });

  test('should render protection page link in header navbar', () => {
    const navProtectionPage = screen.getByTestId('nav-protection-page');
    expect(navProtectionPage).toBeInTheDocument();
    expect(navProtectionPage).toMatchSnapshot();
  });
});
