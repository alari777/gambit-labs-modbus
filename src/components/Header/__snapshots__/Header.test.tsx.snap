// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header should render home link in header navbar 1`] = `
<a
  class="nav-item nav-link active"
  data-testid="nav-home"
  href="/"
>
  Home
</a>
`;

exports[`should render header should render protection page link in header navbar 1`] = `
<a
  class="nav-item nav-link"
  data-testid="nav-protection-page"
  href="/protection-page"
>
  Protection page (?)
</a>
`;
